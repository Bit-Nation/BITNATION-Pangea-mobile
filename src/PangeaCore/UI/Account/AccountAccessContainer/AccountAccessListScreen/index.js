// @flow

import React from 'react';
import { View, FlatList, Image, Animated } from 'react-native';
import { Button, Text } from 'native-base';
import BackgroundImage from 'pangea-common-reactnative/UI/BackgroundImage';
// import Button from '../../../../components/common/Button';
import styles from './styles';
import FakeNavigationBar from 'pangea-common-reactnative/UI/FakeNavigationBar';
import ScreenTitle from 'pangea-common-reactnative/UI/ScreenTitle';
import ListItem from 'pangea-common-reactnative/UI/ListItem';
import type { Account } from 'pangea-common/types/accounts-types';
import i18n from 'pangea-common/i18n';
import BitnationCover from 'pangea-common-reactnative/assets/images/BitnationCover.png';
import bitnationIntro from 'pangea-common-reactnative/UI/assets/images/bitnationIntro.gif';
import Images from 'pangea-common-reactnative/assets/AssetsImages';

type Props = {
  /**
   * @desc Array of accounts to be displayed.
   */
  accounts: Array<Account>,
  /**
   * @desc Function to be called when an item is selected from the list
   * @param id ID of the account to be opened
   */
  onSelectItem: (id: string) => void,
  /**
   * @desc Function to create an account
   */
  onCreateAccount: () => void,
  /**
   * @desc Function to restore an account
   */
  onRestoreAccount: () => void,
};

const AccountAccessListScreen = ({
  accounts,
  onSelectItem,
  onRestoreAccount,
  onCreateAccount,
}: Props) => (
  <Animated.View style={styles.profilesScreenContainer}>
    <BackgroundImage source={Images.backgroundLogin} />
    {/* <FakeNavigationBar />
    <ScreenTitle title={i18n.t('screens.accounts.title')} /> */}
    <Image source={BitnationCover} style={styles.image} />
    <FlatList
      renderItem={(item) => {
        const account = item.item;
        return (
          <ListItem
            text={account.name}
            textStyle={styles.textStyle}
            onPress={id => onSelectItem(id)}
            id={account.id}
          />
        );
      }}
      keyExtractor={item => item.id}
      data={accounts}
      style={styles.sectionList}
      ItemSeparatorComponent={() => <View style={styles.itemSeparator} />}
    />
    <View style={styles.buttonListContainer}>
      <Button
        block
        style={styles.restoreAccountButton}
        onPress={onRestoreAccount}
      >
        <Text style={styles.restoreAccountButtonText}>{i18n.t('screens.accounts.restoreAccount').toUpperCase()}</Text>
      </Button>

      <Button
        block
        onPress={onCreateAccount}
        style={styles.newAccountButton}
      >
        <Text style={styles.restoreAccountButtonText}>{i18n.t('screens.accounts.newAccount').toUpperCase()}</Text>
      </Button>
      <View style={styles.spacer} />
    </View>

  </Animated.View>
);

export default AccountAccessListScreen;
