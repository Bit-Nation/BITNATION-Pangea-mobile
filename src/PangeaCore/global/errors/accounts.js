export class LoginFailedError extends Error {
  transKey: string = 'loginFailed';
}

export class InvalidPasswordError extends Error {
  transKey: string = 'invalidPassword';
}
