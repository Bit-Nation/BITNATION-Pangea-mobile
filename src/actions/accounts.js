// @flow

import { type Account } from '../types/Account';
import type { AsyncTask } from '../utils/asyncTask';
import type { Mnemonic } from '../types/Mnemonic';

export type AccountsListUpdatedAction = { +type: 'ACCOUNTS_LIST_UPDATED', accounts: Array<Account> };
export type CurrentAccountIdChangedAction = { +type: 'CURRENT_ACCOUNT_ID_CHANGED', currentAccountId: string | null };
export type LoginAction = { +type: 'LOGIN', accountId: string, password: string, deferred: boolean };
export type PerformDeferredLoginAction = { +type: 'PERFORM_DEFERRED_LOGIN' };
export type LoginTaskUpdatedAction = { +type: 'LOGIN_TASK_UPDATED', loginTask: AsyncTask<void> };
export type LogoutAction = { +type: 'LOGOUT' };
export type StartAccountCreationAction = { +type: 'START_ACCOUNT_CREATION' };
export type CheckPinCodeAction = { +type: 'CHECK_PIN_CODE', +pinCode: string, +accountId: string, +callback: (success: boolean) => void };
export type CheckPasswordAction = { +type: 'CHECK_PASSWORD', +password: string, +accountId: string, +callback: (success: boolean) => void };
export type SavePinCodeAction = { +type: 'SAVE_PIN_CODE', +pinCode: string, +accountId: string, +callback: (success: boolean) => void };
export type SavePasswordAction = { +type: 'SAVE_PASSWORD', +password: string, +accountId: string, +callback: (success: boolean) => void };
export type ChangeCreatingAccountFieldAction = { +type: 'CHANGE_CREATING_ACCOUNT_FIELD', +field: string, +value: any };
export type SaveCreatingAccountAction = { +type: 'SAVE_CREATING_ACCOUNT', +callback: (success: boolean) => void };
export type StartRestoreAccountUsingMnemonicAction = { +type: 'START_RESTORE_ACCOUNT_USING_MNEMONIC', +mnemonic: Mnemonic };
export type MnemonicConfirmedAction = { +type: 'MNEMONIC_CONFIRMED', +callback: (success: boolean) => void };

export type Action =
  | AccountsListUpdatedAction
  | CurrentAccountIdChangedAction
  | LoginAction
  | PerformDeferredLoginAction
  | LoginTaskUpdatedAction
  | LogoutAction
  | StartAccountCreationAction
  | CheckPinCodeAction
  | CheckPasswordAction
  | SavePinCodeAction
  | SavePasswordAction
  | ChangeCreatingAccountFieldAction
  | SaveCreatingAccountAction
  | StartRestoreAccountUsingMnemonicAction
  | MnemonicConfirmedAction;

export const ACCOUNTS_LIST_UPDATED = 'ACCOUNTS_LIST_UPDATED';
export const CURRENT_ACCOUNT_ID_CHANGED = 'CURRENT_ACCOUNT_ID_CHANGED';
export const LOGIN = 'LOGIN';
export const LOGIN_TASK_UPDATED = 'LOGIN_TASK_UPDATED';
export const LOGOUT = 'LOGOUT';
export const START_ACCOUNT_CREATION = 'START_ACCOUNT_CREATION';
export const CHECK_PIN_CODE = 'CHECK_PIN_CODE';
export const CHECK_PASSWORD = 'CHECK_PASSWORD';
export const SAVE_PIN_CODE = 'SAVE_PIN_CODE';
export const SAVE_PASSWORD = 'SAVE_PASSWORD';
export const CHANGE_CREATING_ACCOUNT_FIELD = 'CHANGE_CREATING_ACCOUNT_FIELD';
export const SAVE_CREATING_ACCOUNT = 'SAVE_CREATING_ACCOUNT';
export const PERFORM_DEFERRED_LOGIN = 'PERFORM_DEFERRED_LOGIN';
export const START_RESTORE_ACCOUNT_USING_MNEMONIC = 'START_RESTORE_ACCOUNT_USING_MNEMONIC';
export const MNEMONIC_CONFIRMED = 'MNEMONIC_CONFIRMED';

/**
 * @desc Action creator for an action that is called when accounts list updated.
 * @param {Account[]} accounts Updated version of accounts.
 * @returns {AccountsListUpdatedAction} An action.
 */
export function accountListUpdated(accounts: Array<Account>): AccountsListUpdatedAction {
  return {
    type: ACCOUNTS_LIST_UPDATED,
    accounts,
  };
}


/**
 * @desc Action creator for an action that is called when current account id is changed.
 * That happens on login and logout.
 * @param {string | null} currentAccountId New id to be set as current or null if user logged out completely.
 * @return {CurrentAccountIdChangedAction} An action.
 */
export function currentAccountIdChanged(currentAccountId: string | null): CurrentAccountIdChangedAction {
  return {
    type: CURRENT_ACCOUNT_ID_CHANGED,
    currentAccountId,
  };
}

/**
 * @desc Action creator for an action that is called to perform a login.
 * @param {string} accountId Id of account to login to.
 * @param {string} password Password to login.
 * @param {boolean} deferred Flag if login should wait for performDeferredLogin action to proceed.
 * @return {LoginAction} An action.
 */
export function login(accountId: string, password: string, deferred: boolean = false): LoginAction {
  return {
    type: LOGIN,
    accountId,
    password,
    deferred,
  };
}

/**
 * @desc Action creator for an action to perform login that was deferred. That is used to not store password while creating account.
 * @return {PerformDeferredLoginAction} An action.
 */
export function performDeferredLogin(): PerformDeferredLoginAction {
  return {
    type: PERFORM_DEFERRED_LOGIN,
  };
}

/**
 * @desc Action creator for an action that is called when login task status updated.
 * @param {AsyncTask<void>} loginTask Updated async task.
 * @return {LoginTaskUpdatedAction} An action.
 */
export function loginTaskUpdated(loginTask: AsyncTask<void>): LoginTaskUpdatedAction {
  return {
    type: LOGIN_TASK_UPDATED,
    loginTask,
  };
}

/**
 * @desc Action creator for an action that is called to perform a logout.
 * @return {LogoutAction} An action.
 */
export function logout(): LogoutAction {
  return {
    type: LOGOUT,
  };
}


/**
 * @desc Action creator for an action that is called to start creation of new account.
 * @return {StartAccountCreationAction} An action.
 */
export function startAccountCreation(): StartAccountCreationAction {
  return {
    type: START_ACCOUNT_CREATION,
  };
}

/**
 * @desc Action creator for an action that is called to check entered pin code.
 * @param {string} pinCode Pin code to check.
 * @param {string} accountId Id of account which pin code is going to be checked.
 * @param {function} callback Callback that is called with true if check is successful and false otherwise.
 * @return {CheckPinCodeAction} An action.
 */
export function checkPinCode(pinCode: string, accountId: string, callback: (boolean) => void): CheckPinCodeAction {
  return {
    type: CHECK_PIN_CODE,
    accountId,
    pinCode,
    callback,
  };
}

/**
 * @desc Action creator for an action that is called to check entered password.
 * @param {string} password Password to check.
 * @param {string} accountId Id of account which password is going to be checked.
 * @param {function} callback Callback that is called with true if check is successful and false otherwise.
 * @return {CheckPinCodeAction} An action.
 */
export function checkPassword(password: string, accountId: string, callback: (boolean) => void): CheckPasswordAction {
  return {
    type: CHECK_PASSWORD,
    accountId,
    password,
    callback,
  };
}

/**
 * @desc Action creator for an action that is called to save new pin code.
 * @param {string} pinCode Pin code to save.
 * @param {string} accountId Id of account which pin code is going to be saved.
 * @param {function} callback Callback that is called with true if save is successful and false otherwise.
 * @return {SavePinCodeAction} An action.
 */
export function savePinCode(pinCode: string, accountId: string, callback: (boolean) => void): SavePinCodeAction {
  return {
    type: SAVE_PIN_CODE,
    accountId,
    pinCode,
    callback,
  };
}

/**
 * @desc Action creator for an action that is called to save new password.
 * @param {string} password Password to save.
 * @param {string} accountId Id of account which password is going to be saved.
 * @param {function} callback Callback that is called with true if save is successful and false otherwise.
 * @return {SavePinCodeAction} An action.
 */
export function savePassword(password: string, accountId: string, callback: (boolean) => void): SavePasswordAction {
  return {
    type: SAVE_PASSWORD,
    accountId,
    password,
    callback,
  };
}

/**
 * @desc Action creator for an action that is called to change a field on creating account.
 * @param {string} field Name of the field to change.
 * @param {any} value New value of the field.
 * @return {ChangeCreatingAccountFieldAction} An action.
 */
export function changeCreatingAccountField(field: string, value: any): ChangeCreatingAccountFieldAction {
  return {
    type: CHANGE_CREATING_ACCOUNT_FIELD,
    field,
    value,
  };
}

/**
 * @desc Action creator for an action that is called to save currently created account.
 * @param {Function} callback Callback that receives boolean value - whether save is successful.
 * @return {SaveCreatingAccountAction} An action.
 */
export function saveCreatingAccount(callback: (boolean) => void): SaveCreatingAccountAction {
  return {
    type: SAVE_CREATING_ACCOUNT,
    callback,
  };
}

/**
 * @desc Action creator for an action that is called to restore account using mnemonic.
 * @param {Mnemonic} mnemonic Mnemonic to be used.
 * @return {StartRestoreAccountUsingMnemonicAction} An action.
 */
export function startRestoreAccountUsingMnemonic(mnemonic: Mnemonic): StartRestoreAccountUsingMnemonicAction {
  return {
    type: START_RESTORE_ACCOUNT_USING_MNEMONIC,
    mnemonic,
  };
}

/**
 * @desc Action creator for an action that is called when mnemonic is confirmed to record that to database.
 * @param {Function} callback Callback that receives boolean value - whether save is successful.
 * @return {MnemonicConfirmedAction} An action.
 */
export function mnemonicConfirmed(callback: (boolean) => void): MnemonicConfirmedAction {
  return {
    type: MNEMONIC_CONFIRMED,
    callback,
  };
}
