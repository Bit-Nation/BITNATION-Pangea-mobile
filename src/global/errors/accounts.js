export class AccountCreationError extends Error {}

export class LoginFailedError extends Error {}

export class InvalidPasswordError extends LoginFailedError {}
