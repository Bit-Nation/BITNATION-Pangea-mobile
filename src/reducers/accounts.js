/* eslint-disable no-use-before-define */
// @flow

import _ from 'lodash';
import uuid from 'uuid4';

import {
  type Action as AccountsAction,
  ACCOUNTS_LIST_UPDATED,
  CURRENT_ACCOUNT_ID_CHANGED,
  LOGIN_TASK_UPDATED,
  START_ACCOUNT_CREATION,
  CHANGE_CREATING_ACCOUNT_FIELD,
  START_RESTORE_ACCOUNT_USING_MNEMONIC,
  SAVE_CREATING_ACCOUNT, SET_CURRENT_ACCOUNT_IDENTITY_KEY,
} from '../actions/accounts';
import type { Account, PartialAccount } from '../types/Account';
import TaskBuilder, { type AsyncTask } from '../utils/asyncTask';
import {
  type Action as ProfileAction,
  CANCEL_ACCOUNT_EDITING,
  CHANGE_EDITING_ACCOUNT,
  DONE_ACCOUNT_EDITING,
  START_ACCOUNT_EDITING,
  SET_PUBLIC_KEY,
} from '../actions/profile';
import type { Mnemonic } from '../types/Mnemonic';

export type State = {
  +editingAccount: Account | PartialAccount | null,
  +creatingAccount: PartialAccount | null,
  +currentAccountId: string | null,
  +login: AsyncTask<void>,
  +logout: AsyncTask<void>,
  +accounts: Array<Account>,
  +currentCreation: | { type: 'create' }
    | { type: 'restore', mnemonic: Mnemonic }
    | null,
  +currentAccountIdentityKey: string | null;
};

export const buildEmptyAccount = (): PartialAccount => ({
  id: uuid(),
  name: null,
  location: null,
  avatar: null,
  accountStore: null,
  networkType: 'main',
});

export const initialState: State = {
  editingAccount: null,
  creatingAccount: null,
  currentAccountId: null,
  login: TaskBuilder.empty(),
  logout: TaskBuilder.empty(),
  accounts: [],
  currentCreation: null,
  currentAccountIdentityKey: null,
};

/**
 * @desc Profile reducer.
 * @param {State} state Current state.
 * @param {AccountsAction|ProfileAction} action Performed action.
 * @returns {State} Next state.
 */
export default (state: State = initialState, action: AccountsAction | ProfileAction): State => {
  switch (action.type) {
    case ACCOUNTS_LIST_UPDATED:
      return {
        ...state,
        accounts: action.accounts,
      };
    case CURRENT_ACCOUNT_ID_CHANGED:
      return {
        ...state,
        currentAccountId: action.currentAccountId,
        currentAccountIdentityKey: action.currentAccountId === null ? null : state.currentAccountIdentityKey,
      };
    case SET_CURRENT_ACCOUNT_IDENTITY_KEY:
      return {
        ...state,
        currentAccountIdentityKey: action.identityKey,
      };
    case LOGIN_TASK_UPDATED:
      return {
        ...state,
        login: action.loginTask,
      };

    case START_ACCOUNT_EDITING:
      return {
        ...state,
        editingAccount: getCurrentAccount(state),
      };
    case CHANGE_EDITING_ACCOUNT:
      return {
        ...state,
        editingAccount: action.account,
      };
    case CANCEL_ACCOUNT_EDITING:
      return {
        ...state,
        editingAccount: null,
      };
    case DONE_ACCOUNT_EDITING: {
      return {
        ...state,
        editingAccount: null,
      };
    }
    case START_ACCOUNT_CREATION: {
      const newAccount = buildEmptyAccount();
      return {
        ...state,
        currentCreation: { type: 'create' },
        creatingAccount: newAccount,
        editingAccount: newAccount,
      };
    }
    case CHANGE_CREATING_ACCOUNT_FIELD: {
      return {
        ...state,
        creatingAccount: {
          ...state.creatingAccount,
          [action.field]: action.value,
        },
      };
    }
    case START_RESTORE_ACCOUNT_USING_MNEMONIC: {
      const newAccount = buildEmptyAccount();
      return {
        ...state,
        currentCreation: { type: 'restore', mnemonic: action.mnemonic },
        creatingAccount: newAccount,
        editingAccount: newAccount,
      };
    }
    case SAVE_CREATING_ACCOUNT: {
      return {
        ...state,
        currentCreation: null,
      };
    }
    case SET_PUBLIC_KEY: {
      return {
        ...state,
        publicKey: action.publicKey,
      };
    }
    default:
      return state;
  }
};

export const getCurrentAccount = (state: State) =>
  _.find(state.accounts, account => account.id === state.currentAccountId) || null;

export const isCreatingAccount = (state: State) =>
  state.currentAccountId === null && state.creatingAccount !== null;
