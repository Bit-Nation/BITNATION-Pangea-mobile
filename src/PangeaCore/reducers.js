// @flow

import { combineReducers } from 'redux';
import accounts, { type State as AccountsState } from '@pangea/accounts/accounts-reducers';
import nations, { type State as NationsState } from '@pangea/nations/nations-reducers';
import modifyNation, { type State as ModifyNationState } from '@pangea/nations/modifyNation-reducers';
import wallet, { type State as WalletState } from '@pangea/wallet/wallet-reducers';
import key, { type State as KeyState } from '@pangea/key/key-reducers';
import activity, { type State as ActivityState } from '@pangea/activity/activity-reducers';
import testingMode, { type State as TestingModeState } from '@pangea/accounts/testingMode-reducers';
import chat, { type State as ChatState } from '@pangea/chat/chat-reducers';
import settings, { type State as SettingsState } from '@pangea/settings/settings-reducers';
import dApps, { type State as DAppsState } from '@pangea/dApps/dApps-reducers';
import documents, { type State as DocumentsState } from '@pangea/documents/documents-reducers';
import contacts, { type State as ContactsState } from '@pangea/contacts/contacts-reducers';

export type State = {
  accounts: AccountsState,
  nations: NationsState,
  modifyNation: ModifyNationState,
  wallet: WalletState,
  key: KeyState,
  activity: ActivityState,
  testingMode: TestingModeState,
  chat: ChatState,
  settings: SettingsState,
  dApps: DAppsState,
  documents: DocumentsState,
  contacts: ContactsState,
}

export const subReducers = {
  accounts,
  nations,
  modifyNation,
  wallet,
  key,
  activity,
  testingMode,
  chat,
  settings,
  dApps,
  documents,
  contacts,
};

export default combineReducers(subReducers);
