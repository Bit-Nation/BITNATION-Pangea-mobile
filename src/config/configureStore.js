// @flow

import { createStore, applyMiddleware, compose, type Store } from 'redux';
import logger from 'redux-logger';
import createSagaMiddleware from 'redux-saga';
import createSagaMonitor from '@clarketm/saga-monitor';
import reducers from '../reducers';
import rootSaga from '../sagas';

/**
 * @desc Configures a Redux store.
 * @return {Store} Created store object.
 */
export default function configureStore(): Store {
  const sagaMonitor = createSagaMonitor({
    level: 'log',
    actionDispatch: true,
  });
  const sagaMiddleware = createSagaMiddleware({ sagaMonitor });

  const enhancer = compose(
    applyMiddleware(sagaMiddleware),
    applyMiddleware(logger),
  );

  const store = createStore(reducers, enhancer);
  sagaMiddleware.run(rootSaga);
  return store;
}
