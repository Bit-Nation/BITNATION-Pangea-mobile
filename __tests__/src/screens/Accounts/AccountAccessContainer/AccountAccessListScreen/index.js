import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import { initialState } from '../../../../../../src/reducers/accounts';
import AccountAccessListScreen from '../../../../../../src/screens/Accounts/AccountAccessContainer/AccountAccessListScreen';
import navigatorMock from '../../../../../../__mocks__/Navigator';

test('AccountAccessListScreen renders correctly', () => {
  const initialStateMock = {
    key: initialState,
  };
  const storeMock = configureStore([]);
  const propsMock = {
    navigator: navigatorMock,
  };

  const wrapper = shallow((
    <AccountAccessListScreen {...propsMock} store={storeMock(initialStateMock)} />
  ));
  expect(wrapper).toMatchSnapshot();
});
