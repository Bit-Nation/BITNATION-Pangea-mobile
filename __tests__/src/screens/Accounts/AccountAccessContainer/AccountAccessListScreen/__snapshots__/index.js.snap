// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountAccessListScreen renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AccountAccessListScreen
    navigator={
      Object {
        "pop": [MockFunction],
        "push": [MockFunction],
        "setButtons": [MockFunction],
        "setOnNavigatorEvent": [MockFunction],
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <BackgroundImage
          source={1}
          style={undefined}
        />,
        <FakeNavigationBar
          navBarHidden={false}
        />,
        <ScreenTitle
          title="Accounts"
        />,
        <FlatList
          ItemSeparatorComponent={[Function]}
          data={undefined}
          disableVirtualization={false}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={1}
          onEndReachedThreshold={2}
          renderItem={[Function]}
          scrollEventThrottle={50}
          style={
            Object {
              "flex": 1,
            }
          }
          updateCellsBatchingPeriod={50}
          windowSize={21}
        />,
        <View
          style={
            Object {
              "borderWidth": 0,
              "marginBottom": 40,
              "marginTop": 10,
            }
          }
        >
          <Button
            enabled={true}
            onPress={[Function]}
            style={
              Object {
                "alignSelf": "stretch",
                "backgroundColor": "transparent",
                "borderRadius": 0,
              }
            }
            styleTitle={
              Object {
                "color": "#58595B",
                "fontSize": 15,
                "fontWeight": "bold",
              }
            }
            testID="accountAccess_restoreButton"
            title="Restore Account"
          />
          <View
            style={undefined}
          />
        </View>,
        <Button
          enabled={true}
          onPress={[Function]}
          style={
            Object {
              "alignSelf": "stretch",
              "backgroundColor": "white",
              "borderRadius": 0,
              "height": 54,
            }
          }
          styleTitle={
            Object {
              "color": "#FF8B00",
              "fontSize": 15,
              "fontWeight": "900",
            }
          }
          testID="accountAccess_createButton"
          title="NEW ACCOUNT"
        />,
      ],
      "style": Object {
        "alignItems": "stretch",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "flex-start",
      },
      "testID": "accountAccess_wrapperView",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "source": 1,
          "style": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "navBarHidden": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "title": "Accounts",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "ItemSeparatorComponent": [Function],
          "data": undefined,
          "disableVirtualization": false,
          "horizontal": false,
          "initialNumToRender": 10,
          "keyExtractor": [Function],
          "maxToRenderPerBatch": 10,
          "numColumns": 1,
          "onEndReachedThreshold": 2,
          "renderItem": [Function],
          "scrollEventThrottle": 50,
          "style": Object {
            "flex": 1,
          },
          "updateCellsBatchingPeriod": 50,
          "windowSize": 21,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Button
              enabled={true}
              onPress={[Function]}
              style={
                Object {
                  "alignSelf": "stretch",
                  "backgroundColor": "transparent",
                  "borderRadius": 0,
                }
              }
              styleTitle={
                Object {
                  "color": "#58595B",
                  "fontSize": 15,
                  "fontWeight": "bold",
                }
              }
              testID="accountAccess_restoreButton"
              title="Restore Account"
            />,
            <View
              style={undefined}
            />,
          ],
          "style": Object {
            "borderWidth": 0,
            "marginBottom": 40,
            "marginTop": 10,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": undefined,
              "enabled": true,
              "onPress": [Function],
              "style": Object {
                "alignSelf": "stretch",
                "backgroundColor": "transparent",
                "borderRadius": 0,
              },
              "styleTitle": Object {
                "color": "#58595B",
                "fontSize": 15,
                "fontWeight": "bold",
              },
              "testID": "accountAccess_restoreButton",
              "title": "Restore Account",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "style": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": undefined,
          "enabled": true,
          "onPress": [Function],
          "style": Object {
            "alignSelf": "stretch",
            "backgroundColor": "white",
            "borderRadius": 0,
            "height": 54,
          },
          "styleTitle": Object {
            "color": "#FF8B00",
            "fontSize": 15,
            "fontWeight": "900",
          },
          "testID": "accountAccess_createButton",
          "title": "NEW ACCOUNT",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <BackgroundImage
            source={1}
            style={undefined}
          />,
          <FakeNavigationBar
            navBarHidden={false}
          />,
          <ScreenTitle
            title="Accounts"
          />,
          <FlatList
            ItemSeparatorComponent={[Function]}
            data={undefined}
            disableVirtualization={false}
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={1}
            onEndReachedThreshold={2}
            renderItem={[Function]}
            scrollEventThrottle={50}
            style={
              Object {
                "flex": 1,
              }
            }
            updateCellsBatchingPeriod={50}
            windowSize={21}
          />,
          <View
            style={
              Object {
                "borderWidth": 0,
                "marginBottom": 40,
                "marginTop": 10,
              }
            }
          >
            <Button
              enabled={true}
              onPress={[Function]}
              style={
                Object {
                  "alignSelf": "stretch",
                  "backgroundColor": "transparent",
                  "borderRadius": 0,
                }
              }
              styleTitle={
                Object {
                  "color": "#58595B",
                  "fontSize": 15,
                  "fontWeight": "bold",
                }
              }
              testID="accountAccess_restoreButton"
              title="Restore Account"
            />
            <View
              style={undefined}
            />
          </View>,
          <Button
            enabled={true}
            onPress={[Function]}
            style={
              Object {
                "alignSelf": "stretch",
                "backgroundColor": "white",
                "borderRadius": 0,
                "height": 54,
              }
            }
            styleTitle={
              Object {
                "color": "#FF8B00",
                "fontSize": 15,
                "fontWeight": "900",
              }
            }
            testID="accountAccess_createButton"
            title="NEW ACCOUNT"
          />,
        ],
        "style": Object {
          "alignItems": "stretch",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "flex-start",
        },
        "testID": "accountAccess_wrapperView",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "source": 1,
            "style": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "navBarHidden": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "title": "Accounts",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "ItemSeparatorComponent": [Function],
            "data": undefined,
            "disableVirtualization": false,
            "horizontal": false,
            "initialNumToRender": 10,
            "keyExtractor": [Function],
            "maxToRenderPerBatch": 10,
            "numColumns": 1,
            "onEndReachedThreshold": 2,
            "renderItem": [Function],
            "scrollEventThrottle": 50,
            "style": Object {
              "flex": 1,
            },
            "updateCellsBatchingPeriod": 50,
            "windowSize": 21,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Button
                enabled={true}
                onPress={[Function]}
                style={
                  Object {
                    "alignSelf": "stretch",
                    "backgroundColor": "transparent",
                    "borderRadius": 0,
                  }
                }
                styleTitle={
                  Object {
                    "color": "#58595B",
                    "fontSize": 15,
                    "fontWeight": "bold",
                  }
                }
                testID="accountAccess_restoreButton"
                title="Restore Account"
              />,
              <View
                style={undefined}
              />,
            ],
            "style": Object {
              "borderWidth": 0,
              "marginBottom": 40,
              "marginTop": 10,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": undefined,
                "enabled": true,
                "onPress": [Function],
                "style": Object {
                  "alignSelf": "stretch",
                  "backgroundColor": "transparent",
                  "borderRadius": 0,
                },
                "styleTitle": Object {
                  "color": "#58595B",
                  "fontSize": 15,
                  "fontWeight": "bold",
                },
                "testID": "accountAccess_restoreButton",
                "title": "Restore Account",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "style": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": undefined,
            "enabled": true,
            "onPress": [Function],
            "style": Object {
              "alignSelf": "stretch",
              "backgroundColor": "white",
              "borderRadius": 0,
              "height": 54,
            },
            "styleTitle": Object {
              "color": "#FF8B00",
              "fontSize": 15,
              "fontWeight": "900",
            },
            "testID": "accountAccess_createButton",
            "title": "NEW ACCOUNT",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
